packages/remastered/src/entry-server.tsx

```diff
@@ -85,36 +96,55 @@ async function onGet({
     foundRouteKeys,
     routesObject
   );
-  const loadedComponents = mapValues(relevantRoutes, (x) => x.component);
-  const loaderContext = new Map<string, unknown>();
+  const loadedComponents = mapValues(relevantRoutes, (x) => ({
+    component: x.component,
+    errorBoundary: x.errorBoundary,
+  }));
+  const loaderContext: RemasteredAppContext["loaderContext"] = new Map();
   const links: AllLinkTags[] = [];
   const headers = new Headers();
+  let erroredRoute: RouteDefinition<EnhancedRoute> | undefined;
 
   for (const relevantRoute of relevantRoutes.values()) {
     if (relevantRoute.loader) {
-      const params = relevantRoute.givenRoute.params;
-      const loader = relevantRoute.loader;
-      const loaderResult = await checkTime(`${relevantRoute.key} loader`, () =>
-        loader({
-          params,
-          request,
-        })
-      );
-      loaderContext.set(relevantRoute.key, loaderResult);
+      try {
+        const params = relevantRoute.givenRoute.params;
+        const loader = relevantRoute.loader;
+        const loaderResult = await checkTime(
+          `${relevantRoute.key} loader`,
+          () =>
+            loader({
+              params,
+              request,
+            })
+        );
+        loaderContext.set(relevantRoute.key, {
+          tag: "ok",
+          value: loaderResult,
+        });
 
-      if (loaderResult === null || loaderResult === undefined) {
-        loaderNotFound = true;
-      }
+        if (loaderResult === null || loaderResult === undefined) {
+          loaderNotFound = true;
+        }
 
-      if (isHttpResponse(loaderResult)) {
-        if (loaderResult.headers.get("Content-Type") === "application/json") {
-          loaderContext.set(relevantRoute.key, await loaderResult.json());
-        } else if (isLoaderJsonResponse) {
-          const serializedResponse = await serializeResponse(loaderResult);
-          loaderContext.set(relevantRoute.key, serializedResponse);
-        } else {
-          return loaderResult;
+        if (isHttpResponse(loaderResult)) {
+          if (loaderResult.headers.get("Content-Type") === "application/json") {
+            loaderContext.set(relevantRoute.key, {
+              tag: "ok",
+              value: await loaderResult.json(),
+            });
+          } else if (isLoaderJsonResponse) {
+            const serializedResponse = await serializeResponse(loaderResult);
+            loaderContext.set(relevantRoute.key, {
+              tag: "ok",
+              value: serializedResponse,
+            });
+          } else {
+            return loaderResult;
+          }
         }
+      } catch (e) {
+        loaderContext.set(relevantRoute.key, { tag: "err", error: String(e) });
```

Comment:
I don't like that. But errors should not propagate to the client. WHAT SHOULD WE DO for good rendering?! damn